src/main/java/org/terasology/web/resources/serverAdmins/AdminPermissionListResource.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.resources.serverAdmins;
+
+import javafx.util.Pair;
```

Comment:
Sorry, I didn't notice before (I assumed `Pair` was a class you made or a class from some of the utility libraries we use, like Guava) - but the sue of this class is problematic.

I was going to test the code (I didn't have the time to do it in the time range between usage of `Pair`s was introduced and today), however compiling failed with a `package javafx.util does not exist`.
So if I remember correctly you are on Windows so it probably works fine there since JavaFX is bundled with the Oracle JDK (or even JRE, I don't remember) installers. However, OpenJDK - which is the Java environment most people uses on Linux (I don't know how the situation is for macOS) - doesn't bundle it (due to licensing issues IIRC) - it's available as a [separate package](http://openjdk.java.net/projects/openjfx/).

We *could* add JavaFX as a requirement (and instruct OpenJDK users to also install OpenJFX), however I wouldn't really like to do that since neither Terasology or FacadeServer are JavaFX applications, and besides the OpenJDK issue I'd prefer not to depend on a full blown GUI framework like JavaFX just to have an implementation of a pair.

There are 3 alternative solutions I can think of:
* Use a `Pair` implementation from one of the libraries already required for Terasology (which we'll always have on the classpath). I think [Guava has one](https://docs.atlassian.com/fugue/1.2.0/apidocs/com/atlassian/fugue/Pair.html), but I'm not sure if we use that Guava extension (as said I'd prefer not to add a new dependency for a pair). [Apache Commons](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/tuple/Pair.html) should have one as well (but I don't remember if we also use commons for Terasology? you may want to have a look at the engine's Gradle files)
* Build your own `Pair` class - should be quite simple (using Java generics)
* Refactor the code to use custom classes instead of pairs

I'd personally prefer 2 or 3 since 1 would lead to problems in case an external library is removed from the engine's dependencies (for example, from a quick search of the main repo seems like Guava is included but not widely used)